misc-scripts/canonical_transcripts/select_canonical_transcripts.pl

```diff
@@ -61,183 +65,233 @@
             $canonical_changes++;
-        } elsif ($new_canonical->dbID != $old_canonical->dbID) {
+        }
+
+        elsif ($new_canonical->dbID != $old_canonical->dbID) {
             no warnings 'uninitialized';
-            printf "%s (%s) changed transcript from %s (%s) to %s (%s)\n",
-                $gene->stable_id,$gene->dbID,$old_canonical->stable_id,$old_canonical->dbID,
-                $new_canonical->stable_id,$new_canonical->dbID;
+            printf
+                "%s (%s) changed transcript from %s (%s) to %s (%s)\n",
+                $gene->stable_id,
+                $gene->dbID,
+                $old_canonical->stable_id,
+                $old_canonical->dbID,
+                $new_canonical->stable_id,
+                $new_canonical->dbID;
+
+            push @change_list, [ $gene->dbID, $new_canonical->dbID ];
             $canonical_changes++;
-            
-            push @change_list,[$gene->dbID,$new_canonical->dbID];
-            
+
             if ($verbose) {
                 printf "Old transcript: [%s,%s,%s,%s,%s,%s,%s]\n",
                     @{ $transcript_selector->encode_transcript($old_canonical) };
                 printf "New transcript: [%s,%s,%s,%s,%s,%s,%s]\n",
                     @{ $transcript_selector->encode_transcript($new_canonical) };
             }
-            
+
             if ($log_fh) {
-                print $log_fh "//\n" or die "Unable to log choices. Cannot write to file";
-                printf $log_fh "Old=[%s,%s,%s,%s,%s,%s,'%s']\n", @{ $transcript_selector->encode_transcript($old_canonical) };
-                printf $log_fh "New=[%s,%s,%s,%s,%s,%s,'%s']\n", @{ $transcript_selector->encode_transcript($new_canonical) };
+                print $log_fh "//\n";
+                printf $log_fh "Old=[%s,%s,%s,%s,%s,%s,'%s']\n",
+                  @{ $transcript_selector->encode_transcript($old_canonical) };
+                printf $log_fh "New=[%s,%s,%s,%s,%s,%s,'%s']\n",
+                  @{ $transcript_selector->encode_transcript($new_canonical) };
             }
         }
-        
     }
 }
 
-print "Canonical transcript alterations: ".$canonical_changes." from ".$total_genes." genes\n";
-if ($log_fh) {$log_fh->close;}
+print
+    "Canonical transcript alterations: $canonical_changes ".
+    "from $total_genes genes\n";
 
 
-## Change database entries.
 
+## Change database entries.
 if ($write) {
     my $gene_update_sql = "UPDATE gene SET canonical_transcript_id = ? where gene_id = ?";
     my $gene_sth = $dba->dbc->prepare($gene_update_sql);
-    
+
     print "Updating database with new canonical transcripts...\n";
     foreach my $change (@change_list) {
-        print "Changin' ".$change->[1]." on ".$change->[0]."\n" if $verbose;
+        print "Changin' ". $change->[1]. " on ". $change->[0]."\n" if $verbose;
         $gene_sth->execute( $change->[1], $change->[0]);
     }
     print "Done\n";
 }
 
+warn "Done\n";
```

Comment:
Now fixed on this PR (sorry):
https://github.com/Ensembl/ensembl/pull/199/files#diff-03f7b81015cecfc399a731e3d81aa4caR259